File: sample/src/main/java/com/github/lany192/box/sample/data/binder/AreaBinder.java
package com.github.lany192.box.sample.data.binder;

import android.view.View;

import androidx.annotation.NonNull;

import com.github.lany192.arch.items.BindingItemBinder;
import com.github.lany192.box.sample.MockUtils;
import com.github.lany192.box.sample.data.bean.Area;
import com.github.lany192.box.sample.databinding.ItemAreaBinding;
import com.github.lany192.dialog.SimpleDialog;
import com.github.lany192.utils.ImageUtils;
import com.hjq.toast.ToastUtils;

public class AreaBinder extends BindingItemBinder<Area, ItemAreaBinding> {

    @Override
    public void onClick(@NonNull BindingHolder<ItemAreaBinding> holder, @NonNull View view, Area area, int position) {
        int count = area.getSubarea() != null ? area.getSubarea().size() : 0;
        SimpleDialog dialog = new SimpleDialog();
        dialog.setTitle("提示");
        dialog.setMessage(area.getName() + count + "个地级市");
        dialog.setRightButton("确定", () -> ToastUtils.show(area.getName()));
        dialog.setLeftButton("取消");
        dialog.show();
    }

    @Override
    public void convert(@NonNull BindingHolder<ItemAreaBinding> holder, Area item) {
        int count = item.getSubarea() != null ? item.getSubarea().size() : 0;
        ImageUtils.show(holder.getBinding().image, MockUtils.getImageUrl());
        holder.getBinding().title.setText(item.getName());
        holder.getBinding().desc.setText("下辖" + count + "个区/市");
    }
}

File: app/demo/src/main/java/com/lany192/box/demo/binder/AreaBinder.kt
package com.lany192.box.demo.binder

import com.github.lany192.arch.items.ItemBinder
import com.github.lany192.dialog.SimpleDialog
import com.github.lany192.extension.load
import com.github.lany192.extension.toast
import com.github.lany192.interfaces.OnSimpleListener
import com.lany192.box.network.data.bean.Area
import com.lany192.box.demo.MockUtils
import com.lany192.box.demo.databinding.ItemAreaBinding

class AreaBinder : ItemBinder<Area, ItemAreaBinding>() {

    override fun onItemClick(binding: ItemAreaBinding, item: Area, position: Int) {
        val count = if (item.subarea != null) item.subarea.size else 0
        SimpleDialog().apply {
            title = "我是标题"
            message = item.name + count + "个地级市"
            rightButton = "确定"
            leftButton = "取消"
            rightClickListener = OnSimpleListener { toast(item.name) }
        }.show()
    }

    override fun convert(binding: ItemAreaBinding, item: Area, position: Int) {
        val count = if (item.subarea != null) item.subarea.size else 0
        binding.image.load(MockUtils.getImageUrl())
        binding.title.text = item.name
        binding.desc.text = "下辖" + count + "个区/市"
    }
}
